controllers/machine_controller.go

```diff
@@ -371,7 +371,7 @@ func (r *MachineReconciler) reconcileDelete(ctx context.Context, cluster *cluste
 
 		var deleteNodeErr error
 		waitErr := wait.PollImmediate(2*time.Second, 10*time.Second, func() (bool, error) {
-			if deleteNodeErr = r.deleteNode(ctx, cluster, m.Status.NodeRef.Name); deleteNodeErr != nil && !apierrors.IsNotFound(deleteNodeErr) {
+			if deleteNodeErr = r.deleteNode(ctx, cluster, m.Status.NodeRef.Name); deleteNodeErr != nil && !apierrors.IsNotFound(errors.Cause(deleteNodeErr)) {
 				return false, nil
```

Comment:
```suggestion
			if deleteNodeErr = r.deleteNode(ctx, cluster, m.Status.NodeRef.Name); apierrors.IsNotFound(errors.Cause(deleteNodeErr)) {
				return true, nil
```

Reply:
Reading a double negative (not NotFound) can be hard to reason about sometimes - what do you think about this instead?

Reply:
err == nil also need to return true. so maybe this ? 
```
if deleteNodeErr = r.deleteNode(ctx, cluster, m.Status.NodeRef.Name); deleteNodeErr == nil || apierrors.IsNotFound(errors.Cause(deleteNodeErr)) {
  return true
}
return false
```
?

Reply:
SGTM